### ISSUE TITLE
Binary filegroup permissions

### ISSUE THREAD

**Issue by tonickkozlov** [thought-machine/please#478]
I'm using please to build a go binary that accesses some static files.

I'm facing the following issue:
there's a rule that encapsulates some static files for a go server:
```
filegroup(
    name="ui",
    srcs=[
        "dist",
        "index.html",
    ],
    binary=True, # marked binary to be seen by resulting go_binary
)
```
This rule is relied on by a `go_binary` rule in parent directory.

The issue arises when re-building the binary filegroup rule:

```
./pleasew build //canary/ui:bin
```
```
    //canary/ui:bin
remove plz-out/bin/canary/ui/dist/main.js: permission denied
```

Looks like the permissions set on the output folder are not correct:
```
ls -l plz-out/bin/canary/ui
total 8
dr-xr-xr-x  3 anton  staff   96 23 Nov 15:48 dist
-r-xr-xr-x  2 anton  staff  505 22 Nov 13:27 index.html
```
Notice lack of write permission on the `dist` directory for the owner.

If the rule would not be marked as binary, the permission would be there:
```
ls -l plz-out/gen/canary/ui
total 8
drwxr-xr-x  3 anton  staff   96 23 Nov 15:53 dist
-r-xr-xr-x  3 anton  staff  505 22 Nov 13:27 index.html
```

Please help me solve the permission issue of figure out what's the right way to produce a filegroup that can be accessed by a go_binary.

Thanks!

**Comment by peterebden** [comment]
`binary` outputs are usually just one file, but that isn't really formally enforced.

In this case I think we are applying permissions incorrectly, we shouldn't remove write permission on directories. Can you take a look please Luna?

**Comment by BNMetrics** [comment]
HI @tonickkozlov ,

This bug has been fixed in #480 ! :)

**Comment by tonickkozlov** [comment]
This is great, thank you!
